request.go

```diff
@@ -1106,3 +1130,35 @@ func (req *ExecuteRequest) Context(ctx context.Context) *ExecuteRequest {
 	req.ctx = ctx
 	return req
 }
+
+// IdRequest informs the server about supported protocol
+// version and features.
+type IdRequest struct {
+	baseRequest
+	idInfo
+}
+
+// newIdRequest returns a new IdRequest.
+func newIdRequest(version ProtocolVersion, features []Feature) *IdRequest {
+	req := new(IdRequest)
+	req.requestCode = IdRequestCode
+	req.version = version
+	req.features = features
+	return req
+}
+
+// Body fills an encoder with the id request body.
+func (req *IdRequest) Body(res SchemaResolver, enc *encoder) error {
+	return fillId(enc, req.version, req.features)
+}
+
+// Context sets a passed context to the request.
+//
+// Pay attention that when using context with request objects,
+// the timeout option for Connection does not affect the lifetime
+// of the request. For those purposes use context.WithTimeout() as
+// the root context.
+func (req *IdRequest) Context(ctx context.Context) *IdRequest {
+	req.ctx = ctx
+	return req
+}
```

Comment:
Looks like it can be transferred to protocol.go

Reply:
But it is the code related to requests. And as far as I know, all requests-related code is in `requests.go`.

Reply:
It seems to me a good idea not to make a dump for requests in the `requests.go`. Specialized requests look better in separate files (see `stream.go`, `prepared.go`).